torch/csrc/autograd/utils/python_arg_parsing.h

```diff
@@ -9,23 +9,23 @@ namespace torch { namespace autograd { namespace utils {
 
 // The parameter allow_copy is to accept copy for Tensor.to (and by proxy
 // PackedSequences.to) but not nn.Module.to.
-inline std::tuple<c10::optional<at::Device>, c10::optional<at::ScalarType>, bool, bool>
+inline std::tuple<c10::optional<at::Device>, c10::optional<at::ScalarType>, bool, bool, c10::optional<at::MemoryFormat>>
   parse_to_conversion(PyObject *args, PyObject *kwargs, bool allow_copy) {
   static PythonArgParser parser({
-    "to(Device device=None, ScalarType dtype=None, bool non_blocking=False, bool copy=False)",
-    "to(ScalarType dtype, bool non_blocking=False, bool copy=False)",
-    "to(Tensor tensor, bool non_blocking=False, bool copy=False)",
+    "to(Device device=None, ScalarType dtype=None, bool non_blocking=False, bool copy=False, *, MemoryFormat? memory_format=None)",
+    "to(ScalarType dtype, bool non_blocking=False, bool copy=False, *, MemoryFormat? memory_format=None)",
+    "to(Tensor tensor, bool non_blocking=False, bool copy=False, *, MemoryFormat? memory_format=None)",
```

Comment:
Ugh, we should have made `non_blocking` and `copy` keyword only too XD

Reply:
Going to be separate PR.